cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,49 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
+# This function gets a command string and invoke it
+# until the command returns an empty string or until timeout
+function retry {
+  local -r tries=$1
+  local -r wait_time=$2
+  local -r action=$3
+  local -r wait_message=$4
+
+  local result=$(eval $action)
+  for i in $(seq $tries); do
+    if [[ -z $result ]] ; then
+      echo "[$i/$tries] $wait_message"
+      sleep $wait_time
+      result=$(eval $action)
+    else
+      return 0
+    fi
+  done
+
+  return 1
```

Comment:
You should return with the `result` code, not mask it.

Reply:
The logic here is simple, checking if the command you passed returns an empty string, if it is we loop continues,
if it dosent return 0 (success).
`retry` mask the result with '1' and '0' so when you call it you wont need to check whether the return value is empty string or not, the code will be cleaner and more straightforward.
Because of how `kubectl` return values are, it is necessary for the command to return an empty string if it fails (using grep for example)
This script is complex already, I am trying not to complicate it even more :)